src/server.mjs

```diff
@@ -1360,6 +1367,47 @@ const timeoutController = (seconds) => {
   return controller;
 }
 
+// Expects a bed url and a chunk name
+// Attempts to download tracks associated with the chunk name from the bed url
+// Returns downloaded tracks as a tracks object
+async function getChunkTracks (bedURL, chunk) {
+  // Download tracks.json file
+  await retrieveChunk(bedURL, chunk, false);
+
+    // Get the path to where the track is downloaded
+    let chunkPath = bedChunkLocalPath(bedURL, chunk);
+    let track_json = path.resolve(chunkPath, "tracks.json");
+    let tracks = null;
+    // Attempt to read tracks.json and covnert it into a tracks object
+    if (fs.existsSync(track_json)) {
+      // Create string of tracks data
+      const string_data = fs.readFileSync(track_json);
+
+      // Convert to object container like the client component prop types expect
+      tracks = JSON.parse(string_data);
+    }
+
+    return tracks;
+}
+
+// Expects a request with a bed url and a chunk name
+// Returns tracks retrieved from getChunkTracks
+api.post("/getChunkTracks", (req, res, next) => {
+  console.log("received request for chunk tracks");
+  if (!req.body.bedURL || !req.body.chunk) {
+    throw new BadReqeustError("Invalid request format", req.body.bedURL, req.body.chunk);
+  }
+  let promise = (async () => {
+    // tracks are falsy if fetch is unsuccessful
+    const tracks = await getChunkTracks(req.body.bedURL, req.body.chunk);
+    res.json({ tracks: tracks });
+  })();
```

Comment:
This asynchronous operation also needs to hold a reader lock on the upload/download directories. Otherwise it isn't safe to call any of the download-and-read functions.